src/operation/in_memory_model_store.jl

```diff
@@ -145,37 +134,41 @@ function initialize_model_storage!(
     @debug "Initialized optimizer_stats_datasets $(get_name(model))"
 end
 
+function list_keys(store::InMemoryModelStore, container_type)
+    container = getfield(store.data, container_type)
+    return collect(keys(container))
+end
+
 function write_result!(
     store::InMemoryModelStore,
     container_type,
-    timestamp,
+    key,
+    execution,
     array,
     columns = nothing,
 )
-    container = getfield(store.data[model_name], container_type)
-    container[name][timestamp] = axis_array_to_dataframe(array, columns)
+    container = getfield(store.data, container_type)
+    df = axis_array_to_dataframe(array, columns)
+
+    # TODO DT: PERF: names are not in the same order.
```

Comment:
Is there a way to ensure that we get the names in the same order at initialization and every call here?

Reply:
I think the names of data frames are always in the same order.